test/arrayops.jl

```diff
@@ -362,6 +362,14 @@ a = [0,1,2,3,0,1,2,3]
 @test findprev(isodd, [2,4,5,3,9,2,0], 7) == 5
 @test findprev(isodd, [2,4,5,3,9,2,0], 2) == 0
 
+# find with general iterables
+s = "julia"
+@test find(s) == [1,2,3,4,5]
```

Comment:
Yes, but returning all valid indices isn't the definition of `find`, so I don't think it's a good idea to use it for that. If needed, we could imagine using the new function from https://github.com/JuliaLang/julia/pull/16260 for that.